charts/simple-app/templates/prometheusrules.yaml

```diff
@@ -0,0 +1,265 @@
+{{ $values           := .Values }}
+{{ $global           := . }}
+{{ $targetNamespace  := .Release.Namespace }}
+{{ $runbookUrl       := required "Values.runbookUrl can not be blank!" .Values.runbookUrl }}
+{{ $appName          := include "simple-app.fullname" . }}
+{{ if .Values.prometheusRules.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: {{ .Release.Name }}-{{ .Chart.Name }}-rules
+  labels:
+    {{- include "simple-app.labels" . | nindent 4 }}
+spec:
+  groups:
+  - name: {{ .Release.Namespace }}.{{ .Release.Name }}.{{ .Chart.Name }}.PodRules
+    rules:
+
+    {{ with .Values.prometheusRules.PodContainerTerminated -}}
+    - alert: PodContainerTerminated
+      annotations:
+        summary: {{`Pod {{$labels.pod}} in namespace {{$labels.namespace}} in error status`}}
+        runbook_url: {{ $runbookUrl }}#kube-pod-container-terminated
+        description: >-
+          Pod {{`{{$labels.pod}}`}} in namespace {{`{{$labels.namespace}}`}}
+          has a container that has been terminated ({{`{{ $value }}`}} times) due to
+          {{`{{$labels.reason}}`}} in the last {{ .for }}.
+      expr: |-
+        sum by (container, namespace, pod, reason) (
+          sum_over_time(
+            kube_pod_container_status_terminated_reason{
+              reason=~"{{ join "|" .reasons }}",
+              namespace="{{ $targetNamespace }}",
+              pod=~"{{ include "simple-app.fullname" $global }}-.*"
```

Comment:
if we are confident this pattern matching will always work, it seems a little sus to me

Reply:
I think it works pretty well.. This is the full name of the deployment. So of course someone could find a way to create another deployment that matches the prefix, but at some point we can't really prevent that. Here's the scenario I could imagine:

  * App1: "myapp"
  * App2: "myapp-async"

In the above case, you could find that the alerts created for "myapp" apply to both "myapp-.*" and "myapp-async-.*" pods. I don't see a good way to get around that. However, as I write this, I can think of another scenario that is more likely to be hit:

Developer intentionally (or not) creates a nearly 63-char long deployment name by setting `.fullnameOverride`: "really-long-string-that-is-too-long-and-may-be-truncated-for-sure". When this happens, the replicaset automatically truncates part of the name:

```
$  k get deployments,rs,pods
NAME                                                                              READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/really-long-string                                                1/1     1            1           6m7s
deployment.apps/really-long-string-that-is-too-long-and-may-be-truncated-for-su   1/1     1            1           5m32s

NAME                                                                                         DESIRED   CURRENT   READY   AGE
replicaset.apps/really-long-string-7f8689d4f                                                 1         1         1       6m7s
replicaset.apps/really-long-string-that-is-too-long-and-may-be-truncated-for-su-5cf866cd98   1         1         1       5m32s

NAME                                                                  READY   STATUS    RESTARTS   AGE
pod/really-long-string-7f8689d4f-tlk4x                                1/1     Running   0          6m7s
pod/really-long-string-that-is-too-long-and-may-be-truncated-fn4z5j   1/1     Running   0          2m1s
```

In this case, we need to handle the fact that the pod-name is going to be truncated to attach a 5-character string at the end... so we need to truncate the prefix for the `pod=~...` selector to 58 characters.